server/routerlicious/packages/lambdas-driver/src/kafka-service/runner.ts

```diff
@@ -22,37 +22,48 @@ export class KafkaRunner implements IRunner {
 
     // eslint-disable-next-line @typescript-eslint/promise-function-async
     public start(): Promise<void> {
-        this.deferred = new Deferred<void>();
+        if (this.deferred) {
+            throw new Error("Already started");
+        }
+
+        const deferred = new Deferred<void>();
+
+        this.deferred = deferred;
 
         process.on("warning", (msg) => {
             console.trace("Warning", msg);
         });
 
         this.factory.on("error", (error) => {
-            this.deferred.reject(error);
```

Comment:
Why do we need to remove the `this.` part?

Reply:
`this.deferred` could be undefined in the context of this lambda function. Using the `deferred` variable makes it guaranteed to not be `undefined`.

Reply:
Ah, I see. `this.deferred` is set on line 31, but by the time the events are called, maybe `this.deferred` would have already been set as `undefined` like in line 67.